src/sentry/shared_integrations/client/base.py

```diff
@@ -304,7 +303,7 @@ def _request(
         self.record_response_for_disabling_integration(resp)
 
         if resp.status_code == 204:
-            return {}
+            return BaseApiResponse(status_code=204)
```

Comment:
I've been meaning to make this change for a while -- but it's not so simple

some things test `if not response: ...` and assume that's a 204 so it needs quite a lot of work to safely unwind this (probably needs some sort of opt-in `dont_special_case_204s=True` which can then be targetted with `@overload` and not return the empty dict here -- but it's quite a bit of work!)

this probably needs some more involved tracing to nuke properly

also callers really shouldn't need `BaseApiResponseX` since it should be one of the two types specifically (based on the `raw_response=...`) -- so I suspect the github client doesn't need this union either and will always get specifically one or the other response type

Reply:
i see... i put the `{}` back then. is it possible to define `__bool__` to determine if an object is truthy? for 204s we could return `False`